src/ztc/api/views/zaken.py

```diff
@@ -107,3 +113,25 @@ class ZaakTypeViewSet(
     }
     concept_related_fields = ["besluittype_set"]
     notifications_kanaal = KANAAL_ZAAKTYPEN
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(detail=True, methods=["post"])
+    def publish(self, request, *args, **kwargs):
+        instance = self.get_object()
+
+        # check related objects
+        besluittypen = instance.besluittype_set.all()
+        informatieobjecttypen = instance.heeft_relevant_informatieobjecttype.all()
+
```

Comment:
can be more optimized:

`instance.besluittype_set.filter(concept=True).exists()`